portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -68,7 +68,6 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     List<String> tagsToSearch = new ArrayList<String>();
  
     blogMessages = putRepliesWithPosts(blogMessages);
-
     tagsToSearch = updateTagsToSearch(tagsToSearch);
```

Comment:
So the reason I said we don't need to return the "tagsToSearch" in updateTagsToSearch() method is that what you return right now is just the reference that we pass here in line 71. Say if you updated it to return "void", it does not affect here at all in the sense that you can continue to use variable |tagsToSearch|, e.g.:

 blogMessages = putRepliesWithPosts(blogMessages);
updateTagsToSearch(tagsToSearch);
...
response.getWriter().println(gson.toJson(sortAndLoadFromBlogHashMap(
    blogMessages, tagsToSearch, numberOfCommentsToDisplay)));
}

Note that passing |tagsToSearch| does not mean you lose the reference within the method. And updating it to "void" makes it less confusing because now readers know you take a reference and potentially update its content.